From 79684305b6cb5a9f79f48fa1c2769dcab0027bd5 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Thu, 2 Apr 2020 07:08:53 -0700
Subject: [PATCH] cog-launcher: Move auto ptr declation before goto statement

This fixes error like

/mnt/b/yoe/build/tmp/work/cortexa7t2hf-neon-vfpv4-yoe-linux-gnueabi/cog/0.5.1-r0/cog-0.5.1/core/cog-launcher.c:529:9: error: cannot jump from this goto statement to its label
goto bad_format;
^
/mnt/b/yoe/build/tmp/work/cortexa7t2hf-neon-vfpv4-yoe-linux-gnueabi/cog/0.5.1-r0/cog-0.5.1/core/cog-launcher.c:531:27: note: jump bypasses initialization of variable with __attribute__((cleanup))
g_autoptr(SoupCookie) cookie = soup_cookie_parse (contents, NULL);
^

Upstream-Status: Submitted [https://github.com/Igalia/cog/pull/199]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 core/cog-launcher.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/core/cog-launcher.c b/core/cog-launcher.c
index 9759a24..10148fc 100644
--- a/core/cog-launcher.c
+++ b/core/cog-launcher.c
@@ -505,6 +505,7 @@ option_entry_parse_cookie_add (const char          *option G_GNUC_UNUSED,
                                GError             **error G_GNUC_UNUSED)
 {
     g_autoptr(GMainLoop) loop = NULL;
+    g_autoptr(SoupCookie) cookie = NULL;
     g_autofree char *domain = g_strdup (value);
 
     char *flagstr = strchr (domain, ':');
@@ -528,7 +529,7 @@ option_entry_parse_cookie_add (const char          *option G_GNUC_UNUSED,
     if (!contents[0])
         goto bad_format;
 
-    g_autoptr(SoupCookie) cookie = soup_cookie_parse (contents, NULL);
+    cookie = soup_cookie_parse (contents, NULL);
     if (!cookie)
         goto bad_format;
 
-- 
2.26.0

